### PR TITLE
added wait time to eventgrid subscriber

### PR DESCRIPTION
**Pull request by adityapawan-b**
added wait time to eventgrid subscriber

Please verify and commit changes accordingy

### CODE REVIEW

File: services/eventgrid/subscription/fn/v1/main.tf [hunk 1]
```diff
@@ -13,4 +13,15 @@ resource "azurerm_eventgrid_event_subscription" "subscription" {
     url = local.webhook_endpoint
   }
 
+}
+
+resource "time_sleep" "wait_90_seconds" {
+  depends_on = [azurerm_eventgrid_topic.topic]
```
Comment (markti-persistent) [inline services/eventgrid/subscription/fn/v1/main.tf:19]: no resource "topic" is declared within the scope of this module. doesn't this panic when you test locally?

Comment (adityapawan-b) [reply services/eventgrid/subscription/fn/v1/main.tf]: corrected the code to point it to topic id

File: services/eventgrid/subscription/fn/v1/main.tf [hunk 2]
```diff
@@ -16,7 +16,7 @@ resource "azurerm_eventgrid_event_subscription" "subscription" {
 }
 
 resource "time_sleep" "wait_90_seconds" {
-  depends_on = [azurerm_eventgrid_topic.topic]
+  depends_on = [azurerm_eventgrid_topic.topic.id]
```
Comment (markti-persistent) [inline services/eventgrid/subscription/fn/v1/main.tf:19]: this won't work either. the problem is that there is no topic object declared in this module. When you say 

depends_on = [azurerm_eventgrid_topic.topic]

you are saying "hey there is an 'azurerm_eventgrid_topic' resource with an object name 'topic' that I depend on".

However, there is no resource of type "azurerm_eventgrid_topic" declared in this module. In fact, this module takes in a STRING input variable called "topic_id". The value of this topic_id variable would likely be set by the caller passing in the topic_id that is output by initializing the 'eventgrid/topic/v1' module...

**Comment by adityapawan-b** [comment]
I will configure WSL on my machine and run locally. Then comeback with proper approach.

Thanks
Aditya pavan
________________________________

**Pull request closed without merging**